SparkCatalogMetaStoreClient.alter_table fails with 'spark.sql.* table property keys not allowed' on Spark-managed Hudi tables

## Bug Description

**What happened:**

`SparkCatalogMetaStoreClient.alter_table` (and `createTable`) fail when the incoming `Table` carries properties starting with `spark.sql.`. Spark's `HiveExternalCatalog.alterTable` / `createTable` rejects such keys with:

```
AnalysisException: Cannot persist <db>.<table> into Hive metastore as table property keys
may not start with 'spark.sql.': [spark.sql.create.version, spark.sql.sources.provider,
spark.sql.sources.schema.partCol.0, spark.sql.sources.schema.numParts,
spark.sql.sources.schema.numPartCols, spark.sql.sources.schema.part.0]
```

This blocks `HoodieHiveSyncClient` whenever the synced table was originally created/touched by Spark (which is the common case for tables with provider=hudi).

**What you expected:**

Hive sync via `SparkCatalogMetaStoreClient` should succeed for any table the user already has in their Spark catalog. The `spark.sql.*` keys are written by Spark itself; round-tripping them through alter_table is a no-op semantically because Spark re-derives them from the `CatalogTable`.

**Steps to reproduce:**

1. Have a Hudi table registered in Spark's Hive metastore (created via `CREATE TABLE ... USING hudi` or any flow that goes through `HiveExternalCatalog`).
2. Configure `hoodie.datasource.hive_sync.use_spark_catalog=true` and `HIVE_SYNC_ENABLED=true`.
3. Write to the table; Hudi triggers `HoodieHiveSyncClient.syncSchema` → `client.alter_table`.
4. `SparkCatalogMetaStoreClient.alter_table` calls `toCatalogTable` (which preserves all parameters including `spark.sql.*`), then `externalCatalog.alterTable` → `AnalysisException`.

**Root cause:**

`SparkCatalogMetaStoreClient.scala:300-323` (`toCatalogTable`) passes `table.getParameters` straight through:

```scala
properties = Option(table.getParameters).map(_.asScala.toMap).getOrElse(Map.empty)
```

Spark's `HiveExternalCatalog.verifyTableProperties` then rejects any key starting with `spark.sql.`. Since `getTable` reads them from HMS (where Spark writes them on createTable), they are present whenever the table was Spark-managed — i.e. essentially always for a Hudi+Spark workflow.

A fix is in PR #<PR_NUMBER>.

## Environment

**Hudi version:** master (verified on `apache/master` as of `8d348cc6b927`)
**Query engine:** Spark (any 3.x)
**Relevant configs:** `hoodie.datasource.hive_sync.use_spark_catalog=true` with `HIVE_SYNC_ENABLED=true`

## Logs and Stack Trace

```
Caused by: org.apache.spark.sql.AnalysisException: Cannot persist
  rawdatatmp.hudi_trips_partitioned_empty_table into Hive metastore as table property keys
  may not start with 'spark.sql.': [spark.sql.create.version,
  spark.sql.sources.provider, spark.sql.sources.schema.partCol.0,
  spark.sql.sources.schema.numParts, spark.sql.sources.schema.numPartCols,
  spark.sql.sources.schema.part.0]
    at org.apache.spark.sql.hive.HiveExternalCatalog.verifyTableProperties(HiveExternalCatalog.scala:...)
    at org.apache.spark.sql.hive.HiveExternalCatalog.doAlterTable(HiveExternalCatalog.scala:...)
    at org.apache.spark.sql.catalyst.catalog.ExternalCatalog.alterTable(ExternalCatalog.scala:...)
    at org.apache.spark.sql.hive.SparkCatalogMetaStoreClient.alter_table(SparkCatalogMetaStoreClient.scala:69)
    at org.apache.hudi.hive.HoodieHiveSyncClient.updateTableProperties(HoodieHiveSyncClient.java:...)
    ... HoodieHiveSyncClient sync flow ...
```

--- a/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hive/SparkCatalogMetaStoreClient.scala
+++ b/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hive/SparkCatalogMetaStoreClient.scala
@@ -302,6 +302,14 @@ class SparkCatalogMetaStoreClient(syncConfig: HiveSyncConfig)
     val dataFields = cols.map(fs => StructField(fs.getName, CatalystSqlParser.parseDataType(fs.getType), nullable = true, Metadata.empty))
     val partitionFields = partCols.map(fs => StructField(fs.getName, CatalystSqlParser.parseDataType(fs.getType), nullable = true, Metadata.empty))
 
+    // Strip "spark.sql.*" properties before handing off to Spark's external catalog.
+    // HiveExternalCatalog.alterTable / createTable rejects such keys ("Cannot persist ...
+    // table property keys may not start with 'spark.sql.'") because they are reserved for
+    // Spark's internal use (provider, schema parts, create version). Spark re-derives and
+    // writes these from the CatalogTable itself, so dropping them on the way in is safe.
+    val tableProperties = Option(table.getParameters).map(_.asScala.toMap).getOrElse(Map.empty)
+      .filterNot { case (k, _) => k.startsWith("spark.sql.") }
+
     CatalogTable(
       identifier = TableIdentifier(tbl, Some(db)),
       tableType = if ("EXTERNAL_TABLE".equalsIgnoreCase(table.getTableType)) CatalogTableType.EXTERNAL else CatalogTableType.MANAGED,
@@ -315,7 +323,7 @@ class SparkCatalogMetaStoreClient(syncConfig: HiveSyncConfig)
       schema = StructType(dataFields ++ partitionFields),
       provider = Some("hudi"),
       partitionColumnNames = partCols.map(_.getName),
-      properties = Option(table.getParameters).map(_.asScala.toMap).getOrElse(Map.empty))
+      properties = tableProperties)
   }
 
   private def fromCatalogTable(table: CatalogTable): Table = {

